langchain-core/src/output_parsers/openai_tools/json_output_functions_parsers.ts

```diff
@@ -0,0 +1,209 @@
+
+  lc_serializable = true;
+
+  argsOnly = true;
+
+  constructor(config?: { argsOnly?: boolean }) {
+    super();
+    this.argsOnly = config?.argsOnly ?? this.argsOnly;
+  }
+
+  /**
+   * Parses the output and returns a string representation of the function
+   * call or its arguments.
+   * @param generations The output of the LLM to parse.
+   * @returns A string representation of the function call or its arguments.
+   */
+  async parseResult(
+    generations: Generation[] | ChatGeneration[]
+  ): Promise<string> {
+    if ("message" in generations[0]) {
+      const gen = generations[0] as ChatGeneration;
+      const functionCall = gen.message.additional_kwargs.function_call;
+      if (!functionCall) {
+        throw new Error(
+          `No function_call in message ${JSON.stringify(generations)}`
+        );
+      }
+      if (!functionCall.arguments) {
+        throw new Error(
+          `No arguments in function_call ${JSON.stringify(generations)}`
+        );
+      }
+      if (this.argsOnly) {
+        return functionCall.arguments;
+      }
+      return JSON.stringify(functionCall);
+    } else {
+      throw new Error(
+        `No message in generations ${JSON.stringify(generations)}`
+      );
+    }
+  }
+}
+
+/**
+ * Class for parsing the output of an LLM into a JSON object. Uses an
+ * instance of `OutputFunctionsParser` to parse the output.
+ */
+export class JsonOutputFunctionsParser<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Output extends Record<string, any> = Record<string, any>
+> extends BaseCumulativeTransformOutputParser<Output> {
+  static lc_name() {
+    return "JsonOutputFunctionsParser";
+  }
+
+  lc_namespace = ["langchain", "output_parsers", "openai_functions"];
+
+  lc_serializable = true;
+
+  outputParser: OutputFunctionsParser;
+
+  argsOnly = true;
+
+  constructor(
+    config?: { argsOnly?: boolean } & BaseCumulativeTransformOutputParserInput
+  ) {
+    super(config);
+    this.argsOnly = config?.argsOnly ?? this.argsOnly;
+    this.outputParser = new OutputFunctionsParser(config);
+  }
+
+  protected _diff(
+    prev: unknown | undefined,
+    next: unknown
+  ): JSONPatchOperation[] | undefined {
+    if (!next) {
+      return undefined;
+    }
+    const ops = compare(prev ?? {}, next);
+    return ops;
+  }
+
+  async parsePartialResult(
+    generations: ChatGeneration[]
+  ): Promise<Output | undefined> {
+    const generation = generations[0];
+    if (!generation.message) {
+      return undefined;
+    }
+    const { message } = generation;
+    const functionCall = message.additional_kwargs.function_call;
+    if (!functionCall) {
+      return undefined;
+    }
+    if (this.argsOnly) {
+      return parsePartialJson(functionCall.arguments);
+    }
+
+    return {
+      ...functionCall,
+      arguments: parsePartialJson(functionCall.arguments),
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } as Record<string, any> as Output;
```

Comment:
cc @jacoblee93 I had to add this cast because `functionCall` does not overlap enough with `Record<string, any>` for some reason...